### PR TITLE
inital work for replacing Fax API

### PR DESCRIPTION
**Pull request by xenein**
works on #2﻿

### CODE REVIEW

File: froide_fax/fax.py [hunk 1]
```diff
@@ -4,28 +4,26 @@
...
 from froide.foirequest.message_handlers import MessageHandler
 from froide.foirequest.models.message import MessageKind
 
 from .pdf_generator import FaxMessagePDFGenerator
-from .utils import (
-    get_media_url, get_status_callback_url, ensure_fax_number
-)
+from .utils import get_media_url, get_status_callback_url, ensure_fax_number
+
+import requests
```
Comment (stefanw) [inline froide_fax/fax.py:14]: I haven't added `isort`, but I try to keep an order with import sections: builtin, django, third-party, froide, relative.

File: froide_fax/fax.py [hunk 2]
```diff
@@ -58,6 +56,34 @@ def get_twilio_fax_data(fax_sid):
     return fax_data._properties
 
 
+def send_fax_telnyx(
```
Comment (stefanw) [inline froide_fax/fax.py:59]: Maybe there should be a wrapper function `send_fax(to, media_url)` that wraps this function and adds the appropriate other arguments from settings.

**Review by stefanw — COMMENTED** [review]
Nice that the transition is straight-forward. Don't forget that the webhook implementation needs to change as well ([Telnyx docs](https://developers.telnyx.com/docs/v2/programmable-fax/receiving-webhooks)). Unfortunately, they make no mention of how to authenticate the webhook request, so we might need to add a secret webhook URL that needs to be statically configured with Telnyx. Or are there other ways to authenticated the request?

**Comment by xenein** [comment]
Newest commit updates requirements (PyNaCL added for signature-verification) and adds a draft for verifying signatures send along web hooks in the new API. 

Receiving hooks and verifying the signatures has been tested in isolation.

currently old and new api share a url for web hook receiving. (lines 48 - 109 can be deleted once API change is complete.)

I'm afraid I forgot to commit after running black and before actually adding changes, again.

File: froide_fax/views.py [hunk 3]
```diff
@@ -35,87 +43,180 @@ def fax_media_url(request, signed):
...
+
+        payload_json = json.loads(request.body)
+
+        # get message object
+        fax_id = None
+        try:
+            fax_id = payload_json.get("data").get("payload").get("fax_id")
+        except:
+            # this key should always exist. we should never end up here
+            return False
```
Comment (stefanw) [inline froide_fax/views.py]: We are in a view function, so we have to return a `HttpResponse`. Returning `False` will trigger an error in Django. If this state is indeed impossible than we should raise a descriptive `Exception` like `ValueError`, so if the impossible happens we have a good stack trace in our error logging.

File: froide_fax/views.py [hunk 4]
```diff
@@ -35,87 +43,180 @@ def fax_media_url(request, signed):
...
-        ProblemReport.objects.report(
+        # prepare signature data for nacl
+        verify_key = VerifyKey(public_key, encoder=Base64Encoder)
+        message = f"{event_timestamp}|".encode("UTF-8") + request.body
+        signature = Base64Encoder.decode(event_signature)
+
+        # verify signature
+        try:
+            verify_key.verify(message, signature=signature)
+        except:
```
Comment (stefanw) [inline froide_fax/views.py]: Blank `except` statements are an anti-pattern and catch too much (not only `Exception`s but also `BaseException`s). Here it should catch `nacl.signing.BadSignatureError`.

**Review by stefanw — COMMENTED** [review]
Some thoughts, but otherwise looks good. I guess we cannot avoid the extra dependency. I will need to see if it needs additional dev packages on Ubuntu due to the C dependency.

File: froide_fax/views.py (hunk 3 above)
Comment (xenein) [reply froide_fax/views.py]: they should all be changed to `ValueError`s now. The `except`-statements got a relevant exception, too. So they are not too broad any longer.

File: froide_fax/views.py (hunk 4 above)
Comment (xenein) [reply froide_fax/views.py]: this is added in the current commit, too.

**Pull request merged**